### PR TITLE
feat(config): move [[idm.users]] to [[query.users]].

### PR DESCRIPTION
**Pull request by youngsofun**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

not in use yet.  so not considering compatibility @Xuanwo.  @flaneur2020 

usage:

```
[query]
....
...

[[query.users]]
name = "u1"
auth_type="no_password"

```
Fixes https://github.com/datafuselabs/databend/issues/7635

### CODE REVIEW

**Review by Xuanwo — APPROVED** [review]
LGTM

File: src/query/config/src/outer_v0.rs [hunk 1]
```diff
@@ -826,6 +821,10 @@ pub struct QueryConfig {
     /// The maximum timeout in milliseconds since the last insert before inserting collected data.
     #[clap(long, default_value = "0")]
     pub async_insert_stale_timeout: u64,
+
+    #[clap(skip)]
+    #[serde(flatten)]
+    pub idm: IDMConfig,
```
Comment (Xuanwo) [inline src/query/config/src/outer_v0.rs:827]: Please flat this struct directly. I'm afraid we can't load this value from env.

Comment (youngsofun) [reply src/query/config/src/outer_v0.rs]: done

**Review by BohuTANG — APPROVED** [review]
+1

**Comment by youngsofun** [comment]
to use double_sha1_password


```
echo -n "password" | sha1sum |cut -d ' ' -f 1| xxd -r -p | sha1sum
2470c0c06dee42fd1618bb99005adca2ec9d1e19  -
```

```
[[query.users]]
name = "user1"
auth_type="double_sha1_password"
auth_string="2470c0c06dee42fd1618bb99005adca2ec9d1e19"
```

**Pull request merged**